### PR TITLE
Fcm

### PR DESCRIPTION
**Pull request by secondsun**
OK this is a bunch of things and I probably need to make a BUNCH of edits and rebasing to make this more review able.

With that said, @danielpassos @corinnekrych @edewit @matzew :eyeglasses: ?

### CODE REVIEW

**Comment by edewit** [comment]
@secondsun it need more work before a review, but you want us to look at it anyway?

**Comment by secondsun** [comment]
@edewit Yes
@matzew Yes.

**Comment by secondsun** [comment]
@matzew @danielpassos @everyoneElse  As promised here are some instructions for testing : 



You will need to deploy this patched version of UPS here : https://drive.google.com/open?id=0BzSy-Z4v4qE-TGozZld4Q1ZyUE0 (matzew++ for the gcm PR).

You can test an FCM enabled app. II have a patched version of the hello world app here : https://github.com/secondsun/unified-push-helloworld/tree/fcm

You will also need follow the instructions for "Import your GCM project as a Firebase project" here : https://developers.google.com/cloud-messaging/android/android-migrate-fcm#import_your_gcm_project_as_a_firebase_project

Finally, you will need to add the google-services.json file.  The firebase console will explain this.

I understand these are a bit scatter shot, but there are a lot of small changes everywhere to keep track of.

**Comment by danielpassos** [comment]
👀

File: aerogear-android-push-test/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -28,11 +28,6 @@
     <uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE" />    
     <uses-permission android:name="com.google.android.c2dm.permission.RECEIVE" />
```
Comment (matzew) [inline aerogear-android-push-test/src/main/AndroidManifest.xml]: let's remove this too ? (see https://developers.google.com/cloud-messaging/android/android-migrate-fcm#remove_the_permissions_required_by_gcm )

**Comment by matzew** [comment]
Does it make sense to fix the build
```
0.99s$ echo no | android create avd --force -n test -t "$ANDROID_EMULATOR" --abi "$ANDROID_ABI"
Valid ABIs: no ABIs.
Error: Invalid --abi armeabi-v7a for the selected target.
```

at some point? Or not really possible on Travis ?

**Comment by matzew** [comment]
@secondsun it compiles :)

will test the app later :)

**Comment by secondsun** [comment]
@matzew Probably makes sense to put some time to fixing the build.

@everyone_else, this works with stock UPS so you don't need my special war

**Comment by secondsun** [comment]
@All Manifest merging is now fixed, so this : https://github.com/jboss-mobile/unified-push-helloworld/pull/31/commits/2a7750ac294db3ae45045a37919c6a8d4d9661ca is all the configuration you need

**Comment by matzew** [comment]
👍 I am fine in merging this

**Comment by secondsun** [comment]
Before I merge, did anyone test topics?

**Comment by matzew** [comment]
let me do one more, though

On Wednesday, 8 June 2016, Matthias Wessendorf <matzew@apache.org> wrote:

> I did, implictly ;)
>
> On Wednesday, 8 June 2016, Hoyt Summers Pittman <notifications@github.com
> <javascript:_e(%7B%7D,'cvml','notifications@github.com');>> wrote:
>
>> Before I merge, did anyone test topics?
>>
>> —
>> You are receiving this because you were mentioned.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/aerogear/aerogear-android-push/pull/62#issuecomment-224555511>,
>> or mute the thread
>> <https://github.com/notifications/unsubscribe/AAJnztbWGrhuB7bknwYcjpn2sxtzFmToks5qJp9EgaJpZM4Iq-o_>
>> .
>>
>
>
> --
> Sent from Gmail Mobile
>


-- 
Sent from Gmail Mobile

**Comment by matzew** [comment]
I did, implictly ;)

On Wednesday, 8 June 2016, Hoyt Summers Pittman <notifications@github.com>
wrote:

> Before I merge, did anyone test topics?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/aerogear/aerogear-android-push/pull/62#issuecomment-224555511>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAJnztbWGrhuB7bknwYcjpn2sxtzFmToks5qJp9EgaJpZM4Iq-o_>
> .
>


-- 
Sent from Gmail Mobile

**Comment by matzew** [comment]
did it, got message. on the server log:
<pre>
==> aerogear-push/logs/server.log <==
2016-06-08 08:05:30,158 INFO  [org.jboss.aerogear.unifiedpush.message.sender.GCMPushNotificationSender] (Thread-2097 (HornetQ-client-global-threads-25476799)) Sent push notification to GCM topic: &#x2F;topics&#x2F;lol
</pre>

**Pull request merged**